src/driver/postgres/PostgresDriver.ts

```diff
@@ -922,7 +933,14 @@ export class PostgresDriver implements Driver {
      */
     protected async createPool(options: PostgresConnectionOptions, credentials: PostgresConnectionCredentialsOptions): Promise<any> {
 
-        credentials = Object.assign({}, credentials, DriverUtils.buildDriverOptions(credentials)); // todo: do it better way
+        credentials = Object.assign({}, credentials);
+
+        if (credentials.url) {
+            credentials = Object.assign(
+                credentials,
+                PostgresDriver.parseConnectionUrl(credentials.url)
```

Comment:
Is it possible to pass the URL as `connectionString` 

See "connection uri" in https://node-postgres.com/features/connecting

Reply:
Well it's possible. But then the method "parseConnectionUrl" I added has no more reason to be, ie. the fix is one line (pass the URL as connectionString). I think it's better to do as you suggest but I am not sure then how to add a test for this.

Reply:
Hmm.. well, less is more!  :)

I'm not sure, though - what happens if you try to open a connection to a socket that doesn't exist in postgres?  Can we confirm the driver handles that as expected?

Reply:
Yes of course, it would be the same as opening a TCP/IP connection that can't be open. I am ready to push this one line fix but the only thing you should be aware is I am not sure how to test this: CI platforms don't provide a way to connect through an UNIX socket (I mean maybe but it's not standard). I think it's not necessary to add a test for this. What do you think? Anyway i will push the one line in a moment.

Reply:
If it's not possible to test it then so be it.  If there's any way we can or if you think of something, let's get more testing coverage.